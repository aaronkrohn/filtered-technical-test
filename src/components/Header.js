import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Wrapper = styled.header`
  background: #f8f8f8;
  padding: 1.875em 0;
`;

const Summary = styled.div`
  max-width: 71.25em;
  padding: 0 1em;
  margin: 0 auto;
`;

const Content = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin: 0 -1em;
`;

const Total = styled.div`
 flex: 0 0 50%;
 font-weight: bold;
 max-wdith: 50%
 padding: 0 1em;
`;

const Breakdown = styled.div`
  flex: 0 0 50%;
 max-wdith: 50%
 padding: 0 1em;
 text-align: right;

 span {
   font-size: 0.85em;
 }
`;

const Header = ({
  numberOfArticles,
  numberOfDocuments,
  numberOfOnlineCourses,
  numberOfVideos,
  totalNumberOfAssets
}) => (
  <Wrapper>
    <Summary>
      <Content>
        <Total>
          {totalNumberOfAssets === 1
            ? '1 asset'
            : `${totalNumberOfAssets} assets`}
        </Total>
        <Breakdown>
          {numberOfArticles > 0 && (
            <Fragment>
              {numberOfArticles} x{' '}
              <span>{numberOfArticles === 1 ? 'Article' : 'Articles'}</span>
              {(numberOfDocuments > 0 ||
                numberOfOnlineCourses > 0 ||
                numberOfVideos > 0) && <Fragment>{', '}</Fragment>}
            </Fragment>
          )}
          {numberOfDocuments > 0 && (
            <Fragment>
              {numberOfDocuments} x{' '}
              <span>{numberOfDocuments === 1 ? 'Document' : 'Documents'}</span>
              {(numberOfOnlineCourses > 0 || numberOfVideos > 0) && (
                <Fragment>{', '}</Fragment>
              )}
            </Fragment>
          )}
          {numberOfOnlineCourses > 0 && (
            <Fragment>
              {numberOfOnlineCourses} x{' '}
              <span>
                {numberOfOnlineCourses === 1
                  ? 'Online Course'
                  : 'Online Courses'}
              </span>
              {numberOfVideos > 0 && <Fragment>{', '}</Fragment>}
            </Fragment>
          )}
          {numberOfVideos > 0 && (
            <Fragment>
              {numberOfVideos} x{' '}
              <span>{numberOfVideos === 1 ? 'Video' : 'Videos'}</span>
            </Fragment>
          )}
        </Breakdown>
      </Content>
    </Summary>
  </Wrapper>
);

Header.propTypes = {
  numberOfArticles: PropTypes.number,
  numberOfDocuments: PropTypes.number,
  numberOfOnlineCourses: PropTypes.number,
  numberOfVideos: PropTypes.number,
  totalNumberOfAssets: PropTypes.number
};

export default Header;
