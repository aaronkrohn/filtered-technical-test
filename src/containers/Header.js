import { connect } from 'react-redux';
import Header from '../components/Header';

const mapStateToProps = ({ previewReducer }) => ({
  numberOfArticles:
    previewReducer.assetList &&
    previewReducer.assetList.length > 0 &&
    previewReducer.assetList.filter(asset => asset.type === 'Article').length
      ? previewReducer.assetList.filter(asset => asset.type === 'Article')
          .length
      : null,
  numberOfDocuments:
    previewReducer.assetList &&
    previewReducer.assetList.length > 0 &&
    previewReducer.assetList.filter(asset => asset.type === 'Document').length
      ? previewReducer.assetList.filter(asset => asset.type === 'Document')
          .length
      : null,
  numberOfOnlineCourses:
    previewReducer.assetList &&
    previewReducer.assetList.length > 0 &&
    previewReducer.assetList.filter(asset => asset.type === 'Online Course')
      .length
      ? previewReducer.assetList.filter(asset => asset.type === 'Online Course')
          .length
      : null,
  numberOfVideos:
    previewReducer.assetList &&
    previewReducer.assetList.length > 0 &&
    previewReducer.assetList.filter(asset => asset.type === 'Video').length
      ? previewReducer.assetList.filter(asset => asset.type === 'Video').length
      : null,
  totalNumberOfAssets: previewReducer.assetList
    ? previewReducer.assetList.length
    : 0
});

export default connect(mapStateToProps)(Header);
