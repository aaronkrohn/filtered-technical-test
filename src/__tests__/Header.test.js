import React from 'react';
import { render } from 'react-testing-library';
import Header from '../components/Header';

test('renders', () => {
  const { container } = render(<Header />);
  expect(container.firstChild).toMatchSnapshot();
});
