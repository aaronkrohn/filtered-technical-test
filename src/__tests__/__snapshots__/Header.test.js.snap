// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<header
  class="sc-bdVaJa ghkLH"
>
  <div
    class="sc-bwzfXH cYJXFx"
  >
    <div
      class="sc-htpNat keCgax"
    >
      <div
        class="sc-bxivhb cRjziJ"
      >
        4 assets
      </div>
      <div
        class="sc-ifAKCX cOHJjj"
      >
        2 x 
        <span>
          Articles
        </span>
        , 1 x 
        <span>
          Documents
        </span>
        , 1 x
        <span>
          Video
        </span>
      </div>
    </div>
  </div>
</header>
`;
